### PR TITLE
Feature/mini cog questionnaire

### PR DESCRIPTION
**Pull request by achen2401**
address #15 
add new Mini-Cog Assessment Questionnaire

Changes include:
- Questionnaire json and CQL for calculating Score
- Tests for verifying CQL calculation

Other Changes:
- update env var `VUE_APP_ALCOHOL_SCREENING_INSTRUMENT` to `VUE_APP_SCREENING_INSTRUMENT` to reflect the general nature of the instrument
- add env var `VUE_APP_FHIR_RESOURCES`, e.g. `Condition,Procedure,Observation`, for more flexible management of what FHIR resources to load
- add optional env var `VUE_APP_TITLE` which allows specification of a custom application title if need to.  It will be displayed in the header.
- Created `launch-context.json` for specifying launch info
- Add header to include patient information, i.e. name, dob

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <link rel="icon" href="<%= BASE_URL %>favicon.png">
     <link href="https://fonts.googleapis.com/css?family=Roboto:100,300,400,500,700,900" rel="stylesheet">
-    <title>Alcohol Screening</title>
+    <title><%= htmlWebpackPlugin.options.title %></title>
```
Comment (pbugni) [inline public/index.html:9]: Nice, thanks for making configurable.

File: public/launch-context.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "clientId": "dummy-client-id",
+    "scope": "patient/Patient.read patient/Condition.read patient/Observation.read patient/Procedure.read patient/QuestionnaireResponse.read online_access openid fhirUser patient/QuestionnaireResponse.write",
```
Comment (pbugni) [inline public/launch-context.json:3]: please add "roles" to this list.

**Review by pbugni — CHANGES_REQUESTED** [review]
Thank you, thank you, for adding all the cql unit tests!
A few minor change requests...

File: test/cases/minicog/minicog_reusable_resources.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+minicog_reusable_resources:
+# patient resource
+- &patientResource
+  resourceType: Patient
+  name: Luke Skywalker
```
Comment (pbugni) [inline test/cases/minicog/minicog_reusable_resources.yml:5]: this is a very different name format than we typically find in FHIR patient resources - but perhaps that's not germane to your testing?

**Review by pbugni — APPROVED** [review]

File: src/components/Survey.vue [hunk 4]
```diff
@@ -56,49 +73,52 @@ export default {
...
+          console.log('Error retrieving FHIR resources ', e);
+          // Send the patient bundle to the CQL web worker WITHOUT FHIR resources
+          sendPatientBundle(this.patientBundle);
+        });
+        this.done();
+        this.ready = true; // We don't show this component until `ready=true`
       }).catch(e => {
         this.error = e;
-        console.log("Patient resource error ", e);
+        console.log("Questionnaire error ", e);
```
[the remark this answers is not in the document]
Comment (achen2401) [reply src/components/Survey.vue]: thanks, this is a questionnaire error. fixed

File: test/cases/minicog/minicog_reusable_resources.yml (hunk 3 above)
Comment (achen2401) [reply test/cases/minicog/minicog_reusable_resources.yml]: thank you for noticing this.  fixed.

File: .env [hunk 5]
```diff
@@ -1,8 +1,32 @@
 VUE_APP_DISPLAY_SCREENING_SCORES=true
-VUE_APP_WRITE_BACK_MODE=none
+
```
Comment (pbugni) [inline .env:2]: We should NOT be committing a `.env` file to the repository.  Please add `.env` to the `.gitignore` and make a `default.env` file as a template, defining the known values and including all that need to be defined.

File: src/Launch.vue [hunk 6]
```diff
@@ -1,33 +1,72 @@
...
       <v-progress-circular
       :value="100"
       indeterminate
       color="primary"
       class="mr-1"></v-progress-circular>
       Loading...
     </div>
+    <div class="pa-8" v-if="error">
+      <v-alert color="error" v-if="error" dark>
+        Error launch the application.
```
Comment (pbugni) [inline src/Launch.vue:13]: "Error launching the application."

**Review by pbugni — CHANGES_REQUESTED** [review]
Awesome progress, thank you @achen2401

**Comment by pbugni** [comment]
@achen2401 looks good, just need the `.env` out of the repo, thanks!

**Review by pbugni — APPROVED** [review]

**Comment by achen2401** [comment]
> @achen2401 looks good, just need the `.env` out of the repo, thanks!

@pbugni wanted to update you on this.  When I did remove the `.env` file, the environment variables are no longer passed to the screen app and the app failed with error.  See screen shot for what environment variables were actually available from browser console as a result:
Screen Shot 2022-06-17 at 9 52 53 AM

 I think the issue may be related to how the [dotenv](https://www.npmjs.com/package/dotenv) library that the frontend build tool, i.e. webpack, uses to read the environment variables.   If I understand correctly from what I read,  dotenv library expects the .env file to be present in the root directory of the screener project when the frontend code **is being built**, i.e. build time vs runtime.  I believed that is changed in a dockerized Safer environment where .env sits at a different level and as a result environment variables are somehow not communicated to the frontend. (Perhaps multiple processes? I am not sure)  I think we had a similar issue in COSRI frontend that we had to workaround (see [here](https://github.com/uwcirg/AHRQ-CDS-Connect-PAIN-MANAGEMENT-SUMMARY/blob/develop/docker-entrypoint.sh#L25))

I have put the .`env` file back for now in case your instance is still being used for demoing

File: .env (hunk 5 above)
Comment (mcjustin) [reply .env:2]: @ivan-c  @achen2401  per our call just now

**Review by mcjustin — APPROVED** [review]

**Pull request merged**